src/rcc.rs

```diff
@@ -318,6 +330,59 @@ impl CFGR {
     }
 }
 
+pub struct LSE {
+    _0: ()
+}
+
+impl LSE {
+    /// Start the LSE clock with a frequency of 32.768 kHz
+    pub fn freeze(self, _bkp: &BackupDomain) -> Lse {
+        // NOTE: Safe because only the BDCR is written to and it is not
+        // used anywhere else.
+        let rcc = unsafe { &*RCC::ptr() };
+
+        rcc.bdcr.modify(|_, w| {
+            w
+                // Enable external low speed oscilator
+                .lseon().set_bit()
+                // Enable the RTC
+                .rtcen().set_bit()
+                // Set the source of the RTC to LSE
+                .rtcsel().lse()
```

Comment:
That's a bit strange to start RTC here. I think we can simplify the LSE thing by just starting it in the RTC constructor.

Reply:
That requires leaking access to BDCR outside RCC, do we want to do that?

Reply:
As BDCR is Backup Domain Control Register, as long as we have access to BackupDomain, that doesn't chock me. But that's right that starting LSE without starting RTC is useless.